modules/apps/adaptive-media/adaptive-media-image-api/src/main/java/com/liferay/adaptive/media/image/counter/BaseAMImageCounter.java

```diff
@@ -50,46 +61,107 @@ public int countExpectedAMImageEntries(long companyId) {
 		return counter.get();
 	}
 
-	protected abstract void forEachFileEntry(
-			long companyId, Consumer<DLFileEntry> consumer)
-		throws PortalException;
-
-	protected String[] getMimeTypes() {
-		return ArrayUtil.filter(
-			amImageMimeTypeProvider.getSupportedMimeTypes(),
-			amImageValidator::isProcessingSupported);
-	}
+	protected abstract long getClassNameId();
 
 	@Reference
 	protected AMImageMimeTypeProvider amImageMimeTypeProvider;
 
 	@Reference
 	protected AMImageValidator amImageValidator;
 
+	@Reference
+	protected ClassNameLocalService classNameLocalService;
+
 	@Reference
 	protected DLFileEntryConfigurationProvider dlFileEntryConfigurationProvider;
 
-	private Consumer<DLFileEntry> _getCountDLFileEntryConsumer(
-		AtomicInteger counter) {
+	@Reference
+	protected DLFileEntryLocalService dlFileEntryLocalService;
+
+	private void _forEachGroup(long companyId, Consumer<Group> groupConsumer)
+		throws PortalException {
+
+		ActionableDynamicQuery actionableDynamicQuery =
+			GroupLocalServiceUtil.getActionableDynamicQuery();
+
+		actionableDynamicQuery.setModelClass(Group.class);
+
+		actionableDynamicQuery.setCompanyId(companyId);
+
+		actionableDynamicQuery.setPerformActionMethod(
+			(ActionableDynamicQuery.PerformActionMethod<Group>)
+				groupConsumer::accept);
+
+		actionableDynamicQuery.performActions();
```

Comment:
The formatting is wrong; you need to sort by property name, and remove unnecessary blank lines between statements. See `com.liferay.change.tracking.internal.dispatch.executor.CTConflictCheckerDispatchTaskExecutor#doExecute` for an example.